c/misra/test/rules/RULE-6-1/gcc/test.c

```diff
@@ -0,0 +1,13 @@
+typedef unsigned int UINT16;
+
+enum Color { R, G, B };
+
+struct SampleStruct {
+  int x1 : 2;          // COMPLIANT
+  unsigned int x2 : 2; // COMPLIANT - explicitly unsigned
+  signed int x3 : 2;   // COMPLIANT - explicitly signed
+  UINT16 x4 : 2;       // COMPLIANT - type alias resolves to a compliant type
+  signed long x5 : 2;  // COMPLIANT
+  signed char x6 : 2;  // COMPLIANT
+  enum Color x7 : 3;   // COMPLIANT
+} sample_struct;
```

Comment:
for self documenting purposes, should we include an extra testcase for the portion of the [gcc doc](https://gcc.gnu.org/onlinedocs/gcc/Structures-unions-enumerations-and-bit-fields-implementation.html) that says - 

"Whether atomic types are permitted for bit-fields (C11 6.7.2.1).
Atomic types are not permitted for bit-fields."

Reply:
I've added the test cases as comments, because this is compiler checked.